webapp-django/crashstats/authentication/views.py

```diff
@@ -49,3 +69,87 @@ def debug_login(request):
     future = datetime.datetime.utcnow() + datetime.timedelta(seconds=10)
     response.set_cookie('cookie_value', cookie_value, expires=future)
     return response
+
+
+@json_view
+def oauth2_signout(request):
+    """If people click the "Sign out" link, an AJAX post will trigger
+    a post to this view. But if the user right-clicks to, for example,
+    "Open Link in New Tab" they'll arrive here.
+
+    If they arrive here, in any way, and are NOT signed in, redirect
+    out.
+    """
+    if not request.user.is_authenticated():
```

Comment:
I guess it doesn't matter. We don't really care about the user being active here.